backend/src/util/common.js

```diff
@@ -12,6 +13,21 @@ function buildFilterQuery(query, mapping) {
   return filterQuery
 }
 
+// Dynamics 365 expects OData queries on dates to be in ISO format
+function formatToISODateFormat(dateString) {
+  if (moment(dateString, 'YYYY-MM-DD', true).isValid()) {
+    return dateString
+  } else {
+    const date = moment(dateString)
+    if (date.isValid()) {
```

Comment:
You could do this cleanly with a ternary expression.

Reply:
Fixed.